add updated property to defintion

dabutvin opened the pull request:
comes through in mongo as 

![image](https://user-images.githubusercontent.com/5168796/49814895-667bc100-fd1f-11e8-9f64-e162d1c48c6e.png)

dabutvin commented:
we can do queries like this

```
db.getCollection('definitions').find({ updated: { $gt: new Date('2018-12-11') } })
```

or more precisesly

```
db.getCollection('definitions').find({ updated: { $gt: new Date('2018-12-11T16:31:00.000Z') } })
```

jeffmcaffer commented on business/definitionService.js:
```diff
@@ -128,6 +128,7 @@ class DefinitionService {
   }
 
   async _store(definition) {
+    definition.updated = new Date()
```
A few questions about this
* should we be adding this to the schema? If the tests are not failing now then the schema is likely wrong. Should not allow additional properties
* In the future we might have more of these sort of `meta` properties. Wanna group them into a parent prop? (e.g., _meta or some such? Suggest we do NOT use _metadata as that will conflict/confuse in some of our store providers)

dabutvin replied:
added to schema and grouped in `_meta` object

dabutvin commented:
re: date

I think we should change to just straight up json.

here's some background if you're interested from [source](https://www.compose.com/articles/from-json-to-bson-and-back/)

> When working with MongoDB dates, ISODate only appears in the shell and can only be consumed by the shell. Everywhere else, Date's are a Date instance. With BSON Dates being stored, date fields become an easily sortable, comparable and indexable number for the database.

It changes the way the data is exported

given:
```
{ "created": new Date("Feb 12 2014 15:28:31 GMT") }
``` 

will get exported as:
```
{
  "created": { "$date": 1392218911000 }
}
```

which is not what we would want

dabutvin commented:
Now the data looks better. Good call

![image](https://user-images.githubusercontent.com/5168796/49965870-4856c200-fed3-11e8-9852-48f22ed627cf.png)

dabutvin commented:
the queries are now

```
db.getCollection('definitions').find({ '_meta.updated': { $gt: '2018-12-13' } })
```

```
db.getCollection('definitions').find({ '_meta.updated': { $gt: '2018-12-13T20:33:00' } })
```

storrisi commented:
Seems like this change of schema has introduced a bug on the UI.
As mentioned in [this doc](https://docs.google.com/document/d/1pB5ue8It7mRO1K4tesLGMTXcbZWVg7Q8N0ibTIft1fA/edit#) from issue #226 , the _meta property was expected to be a different type of object.

I'm gonna fix it with a quick PR on the website.
